CesiumGS/cesium
# Enable eslint-jsdoc

# Description

This PR enables [eslint-plugin-jsdoc](https://github.com/gajus/eslint-plugin-jsdoc). I figured we'd go ahead and enable the defaults, and we'll turn off rules for now that will require a lot of manual fixes. We can systematically enable rules and fix them one-by-one where time allows or need arises. For example, type checking may be particularly helpful.

* Uses the recommended ruleset. In the future, we may want to turn on TS mode, but for now we'll default to the recommend for JS-only.
* Disable rules that are against our coding consistencies, at least for the time being and added comments why
* Disabled quite a few rules which require manual fixes, like filling in types and descriptions, until we can manually fix.
* Autofix with `npm run eslint -- --fix`

## Issue number and link

Fixes https://github.com/CesiumGS/cesium/issues/10815

## Testing plan



1. Run `eslint` and ensure there are no failures
2. Run `build-docs` and ensure the built documentation is not broken
3. Run `build-ts` and ensure there are no errors (there will still be warnings)

# Author checklist
